src/Uno.UWP/Storage/Pickers/FileOpenPicker.wasm.cs

```diff
@@ -0,0 +1,101 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Foundation;
+
+namespace Windows.Storage.Pickers
+{
+	public partial class FileOpenPicker
+	{
+		private const string JsType = "Windows.Storage.Pickers.FileOpenPicker";
+		private const string FileSeparator = "\\\\";
+		private const char FileInfoSeparator = '\\';
+
+		private async Task<StorageFile?> PickSingleFileTaskAsync(CancellationToken token)
+		{
+			var files = await PickFilesAsync(false, token);
+			return files.FirstOrDefault();
+		}
+
+		private async Task<IReadOnlyList<StorageFile>> PickMultipleFilesTaskAsync(CancellationToken token)
+		{
+			return await PickFilesAsync(true, token);
+		}
+
+		private async Task<FilePickerSelectedFilesArray> PickFilesAsync(bool multiple, CancellationToken token)
+		{
+			var showAllEntryParameter = FileTypeFilter.Contains("*") ? "true" : "false";
+			var multipleParameter = multiple ? "true" : "false";
+			var fileTypeMapParameter = BuildFileTypesMap();
+
+			var returnValue = await WebAssemblyRuntime.InvokeAsync($"{JsType}.pickFilesAsync({multipleParameter},{showAllEntryParameter},{fileTypeMapParameter})");
+			var files = returnValue.Split(new string[] { FileSeparator }, StringSplitOptions.RemoveEmptyEntries);
+			var results = new List<StorageFile>();
+			foreach (var file in files)
+			{
+				var fileInfoParts = file.Split(FileInfoSeparator);
+				if (!Guid.TryParse(fileInfoParts[0], out var guid))
+				{
+					throw new InvalidOperationException("Selected file ID is not valid.");
+				}
+				var name = fileInfoParts[1];
+				var contentType = fileInfoParts[2];
+				var storageFile = StorageFile.GetFileFromNativePath(guid, name, contentType);
+				results.Add(storageFile);
+			}
+			return new FilePickerSelectedFilesArray(results.ToArray());
+		}
+
+		private string BuildFileTypesMap()
```

Comment:
Maybe this should be cached?

Reply:
The issue is the file types may change any time, so I would have to add some event to the `FileTypeFilter` to notify about this or something similar. It should not matter too much though, the number of mime extensions is low, so it will not have noticeable impact.